pkg/console/controllers/healthcheck/controller.go

```diff
@@ -150,32 +151,41 @@ func (c *HealthCheckController) Sync(ctx context.Context, controllerContext fact
 }
 
 func (c *HealthCheckController) CheckRouteHealth(ctx context.Context, operatorConfig *operatorsv1.Console, route *routev1.Route) (string, error) {
-	url, _, err := routeapihelpers.IngressURI(route, route.Spec.Host)
-	if err != nil {
-		return "RouteNotAdmitted", fmt.Errorf("console route is not admitted")
-	}
-
-	caPool, err := c.getCA(ctx, route.Spec.TLS)
-	if err != nil {
-		return "FailedLoadCA", fmt.Errorf("failed to read CA to check route health: %v", err)
-	}
-	client := clientWithCA(caPool)
+	var reason string
+	err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
```

Comment:
Trying to understand how this works, [Godocs][1]:

> On a non-"Conflict" error, or if it retries too many times and gives up, RetryOnConflict will return an error to the caller.

I'm not sure if [`Client.Timeout exceeded while awaiting headers`][2] would satisfy [`IsConflict`][6] or not.  And then drilling into [the backing Go][3]:

> backoff defines the maximum retries...

And from [`ExponentialBackoff`][5], the limit seems to be `Steps`.  Which [`DefaultRetry` sets to 5][4].  So other than "will these match [`IsConflict`][6]?", this all makes sense to me.

[1]: https://pkg.go.dev/k8s.io/client-go/util/retry#RetryOnConflict
[2]: https://issues.redhat.com/browse/OCPBUGS-24041
[3]: https://github.com/kubernetes/client-go/blob/v0.29.0/util/retry/util.go#L46
[4]: https://github.com/kubernetes/client-go/blob/v0.29.0/util/retry/util.go#L26-L33
[5]: https://github.com/kubernetes/apimachinery/blob/v0.29.0/pkg/util/wait/backoff.go#L452-L464
[6]: https://github.com/kubernetes/apimachinery/blob/v0.29.0/pkg/api/errors/errors.go#L544-L555

Reply:
@wking good point. Originally I wanted to use `retry.DefaultRetry` but I've ended up using `retry.DefaultBackoff`. 
Reading though the docs I think `retry.DefaultRetry` would be more suitable.
@TheRealJon thoughts ?

Reply:
I agree with @wking that `retry.RetryOnConflict` may not behave as expected. We could probably use `retry.OnError`, which allows us to provide a predicate function to determine if the target function should be retried. I think it would look something like this:

```go
err := retry.OnError(
  retry.DefaultRetry, 
  func (err error) bool { return err != nil }
  func() error { /* health check logic */ }
)